clvm-traits/src/match_byte.rs

```diff
@@ -0,0 +1,45 @@
+use clvmr::{
+    allocator::{NodePtr, SExp},
+    Allocator,
+};
+use num_bigint::{BigInt, Sign};
+
+use crate::{Error, FromClvm, Result, ToClvm};
+
+#[derive(Debug, Copy, Clone)]
+pub struct MatchByte<const BYTE: u8>;
+
+impl<const BYTE: u8> ToClvm for MatchByte<BYTE> {
+    fn to_clvm(&self, a: &mut Allocator) -> Result<NodePtr> {
+        a.new_number(BYTE.into()).map_err(Error::Allocator)
```

Comment:
the allocator has special cases for 0 and 1 (`Allocator::null()` and `Allocator::one()`) which would be more efficient to use for those cases.

Reply:
Added the optimization (it should decide the branch at compile time since the value is a const generic)